graph/impl/KokkosGraph_Distance2Color_impl.hpp

```diff
@@ -616,7 +616,7 @@ class GraphColorDistance2 {
     const lno_t numVerts = this->nr;
     const lno_t numCols  = this->nc;
     // note: relying on forbidden and colors_out being initialized to 0
-    forbidden_view forbidden("Forbidden", batch * numCols);
+    forbidden_view forbidden("Forbidden", static_cast<typename forbidden_view::size_type>(batch) * numCols);
```

Comment:
Getting the size_type for each specific view is correct but really hurts readability. What about just using size_t for these?
```suggestion
    forbidden_view forbidden("Forbidden", size_t(batch) * numCols);
```

Reply:
Are we sure that `size_t` is always the right thing to do? I do not think that in practice there are architectures that provide `long long int` that are really larger than `size_t`?
For this function we could locally define `using size_type = typename forbidden_view::size_type` to make the cast a bit more readable. Regarding the c-style vs. cpp-style cast, I don't think that it's fair to favor one over the other unless we make it a rule for the library?

Reply:
size_t will always be OK when computing anything related to memory, which these examples are. Kokkos hardcodes size_t for extents/strides already (even if a view's size_type is different). It makes sense because size_t can always represent the size of any block of memory.

I don't care that much about ``type(x)`` vs ``(type) x`` vs ``static_cast<type>(x)``, though static_cast is a little verbose and cluttered.

Reply:
I asked @crtrott about this and it turns out ``size_type`` of a view is intended for iterating over a view efficiently, so it may be smaller than ``size_t`` but won't ever be larger.

Reply:
Okay switching to `static_cast<size_t>()` for more clarity